R/db-helpers.R

```diff
@@ -146,11 +146,14 @@ get_src_tbl_names <- function(src, schema = NULL, dbname = NULL) {
   }
 
   names_table %>%
-    filter(schema_name == !!schema) %>%
+    # sql expressions need to be wrapped in parentheses
+    filter(schema_name %in% !!(if (inherits(schema, "sql")) glue_sql_collapse(schema) else schema)) %>%
     collect() %>%
     # create remote names for the tables in the given schema (name is table_name; cannot be duplicated within a single schema)
-    mutate(remote_name = schema_if(schema_name, table_name, con, dbname)) %>%
-    select(-schema_name) %>%
+    transmute(
+      local_name = gsub(" ", "_", if (length(unique(schema_name)) == 1) table_name else paste(schema_name, table_name, sep = ".")),
```

Comment:
What happens if we don't do this? I'd expect an error.

I think this should be opt-in, perhaps with a glue expression that refers to variables named `schema` and `table`?